bricepollock/canyoneer
# Show special favorite pin for favorites on the map and fixup some issues with 'go to on map'

### Description
* If you favorite a pin, show its canyon pin on the map with a "favorite" treatment
* Properly migrate from "last location" to "last viewport" so we can show the last viewport you had more accurately
* Update "Go to canyon on map" to pass in lower zoom so we have a better chance of actually seeing the canyon in heavy cluster areas

<img width="559" alt="Screenshot 2024-12-29 at 4 16 29 PM" src="https://github.com/user-attachments/assets/3adbf9fe-fec0-4a9c-a121-00a630e2494e" />

### Test Plan
* Favorite a pin
* Go to on map
* See the canyon (bug: topo lines do not render until you move the map)
* Tap on a new canyon
* Favorite and dismiss
* Pin updates on map
* Tap on same canyon
* Unfavorite
* Return to map and its back to same pin
